enterprise/models.py

```diff
@@ -107,3 +110,56 @@ def __repr__(self):
         Return uniquely identifying string representation.
         """
         return self.__str__()
+
+
+def logo_path(instance, filename):
+    """
+    Delete the file if it already exist and returns the enterprise customer logo image path.
+
+    :param instance: EnterpriseCustomerBrandingConfiguration object
+    :param filename: file to upload
+    :return path: path of image file e.g. enterprise_logo/1/filename
+    """
+    name = os.path.join('enterprise_logo', str(instance.id), filename)
+    fullname = os.path.join(settings.MEDIA_ROOT, name)
+    if os.path.exists(fullname):
+        os.remove(fullname)
+    return name
+
+
+class EnterpriseCustomerBrandingConfiguration(TimeStampedModel):
+    """
+    Model that keeps track of enterprise branding configurations e.g. enterprise customer logo.
+
+    Fields:
+        enterprise_customer (ForeignKey[EnterpriseCustomer]): enterprise customer
+        logo (ImageField): enterprise customer image
+    """
+
+    enterprise_customer = models.OneToOneField(
+        EnterpriseCustomer,
+        blank=False,
+        null=False,
+        related_name='branding_configuration'
+    )
+    logo = models.ImageField(
+        upload_to=logo_path,
```

Comment:
@bradenmacdonald @mattdrayer what you suggest. **IF** 
The S3 owner create a bucket for enterprise media assets with public read access (access control list), It seems me that any files uploaded through FileField or ImageField attributes on your models should end up in `http://your-bucket-name.s3.amazonaws.com/media/`

Note: AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY is required to access the AWS s3 service.